src/Fantomas.Core/CodePrinter.fs

```diff
@@ -3781,6 +3781,56 @@ and genField astContext prefix (Field (ats, px, ao, isStatic, isMutable, t, so,
         |> optSingle (genTriviaFor SynField_IdentifierAndType) innerRange)
     |> genTriviaFor SynField_ range
 
+and genMultilineAnonRecordType (isStruct: bool) fields (astContext: ASTContext) =
+    let recordExpr =
+        fun (ctx: Context) ->
+            // position after `{| ` or `{|`
+            let targetColumn = ctx.Column + (if ctx.Config.SpaceAroundDelimiter then 3 else 2)
+
+            atCurrentColumn
+                (sepOpenAnonRecd
+                 +> col sepNln fields (fun (AnonRecordFieldType (ident, ty, range)) ->
+                     let expr =
+                         if ctx.Config.IndentSize < 3 then
+                             sepSpaceOrDoubleIndentAndNlnIfExpressionExceedsPageWidth (genType astContext false ty)
+                         else
+                             sepSpaceOrIndentAndNlnIfExpressionExceedsPageWidth (genType astContext false ty)
+
+                     // Add enough spaces to start at the right column but indent from the opening curly brace.
+                     // Use a double indent when using a small indent size to avoid offset warnings.
+                     addFixedSpaces targetColumn
+                     +> atCurrentColumn (enterNodeFor SynExpr_AnonRecd_Field range +> genIdent ident)
+                     +> sepColon
+                     +> expr)
+                 +> sepCloseAnonRecd)
+                ctx
+
+    onlyIf isStruct !- "struct " +> recordExpr
+
+and genMultilineAnonRecordTypeAlignBrackets (isStruct: bool) (fields) astContext =
+    let fieldsExpr = col sepNln fields (genAnonRecordFieldType astContext)
+
+    let genRecord =
+        sepOpenAnonRecdFixed
+        +> indentSepNlnUnindent fieldsExpr
+        +> sepNln
+        +> sepCloseAnonRecdFixed
+
+    ifElse isStruct !- "struct " sepNone
+    +> atCurrentColumnIndent genRecord
+    +> unindent
+
+and genAnonRecordTypeFieldName isStruct fields astContext =
```

Comment:
I would rename this to `genMultilineAnonRecordType` and put it before `genMultilineAnonRecordTypeAlignBrackets`. I think that will be slightly more consistent with how the others are today.

Reply:
There already is a function called `genMultilineAnonRecordType`, what else would you name this one? I agree it's not the most clear name I came up with here haha